client/data/hosting/use-cache.ts

```diff
@@ -0,0 +1,140 @@
+import {
+	useQuery,
+	useMutation,
+	UseMutationOptions,
+	useQueryClient,
+	useIsMutating,
+} from '@tanstack/react-query';
+import { useCallback } from 'react';
+import wp from 'calypso/lib/wp';
+
+export const EDGE_CACHE_ENABLE_DISABLE_NOTICE_ID = 'edge-cache-enable-disable-notice';
+export const USE_EDGE_CACHE_QUERY_KEY = 'edge-cache-key';
+export const TOGGLE_EDGE_CACHE_MUTATION_KEY = 'set-edge-site-mutation-key';
+export const CLEAR_EDGE_CACHE_MUTATION_KEY = 'clear-edge-site-mutation-key';
+
+interface MutationVariables {
+	name: string;
+}
+
+interface MutationResponse {
+	message: string;
+}
+
+interface MutationError {
+	code: string;
+	message: string;
+}
+
+export const getEdgeCacheStatus = async ( siteId: number ) => {
+	const response = await wp.req.get( {
+		path: `/sites/${ siteId }/hosting/edge-cache/active`,
+		apiNamespace: 'wpcom/v2',
+	} );
+
+	return response;
+};
+
+export const setEdgeCache = async ( siteId: number, newStatus: boolean ) => {
+	const response = await wp.req.post( {
+		path: `/sites/${ siteId }/hosting/edge-cache/active`,
+		apiNamespace: 'wpcom/v2',
+		body: {
+			active: newStatus,
+		},
+	} );
+	return response;
+};
+
+export const purgeEdgeCache = async ( siteId: number ) => {
+	return await wp.req.post( {
+		path: `/sites/${ siteId }/hosting/edge-cache/purge`,
+		apiNamespace: 'wpcom/v2',
+	} );
+};
+
+export const useEdgeCacheQuery = ( siteId: number ) => {
+	return useQuery< boolean, unknown, boolean >( {
+		queryKey: [ USE_EDGE_CACHE_QUERY_KEY, siteId ],
+		queryFn: () => getEdgeCacheStatus( siteId ),
+		enabled: !! siteId,
+		select: ( data ) => {
+			return !! data;
+		},
+		meta: {
+			persist: false,
```

Comment:
I know it was already like this before, but wouldn't it be better for this to be `persist=true`? It doesn't feel like this option gets changed all that often and it means the cache card would load instantly on `/hosting-config` page.

Reply:
Eh, I'd be afraid to have the data end up as stale. I think I'm indifferent.